From 64e6c9f12f3999b80d783fb3bb5d469ff6ac9425 Mon Sep 17 00:00:00 2001
From: Sai Surya Teja <sssteja@habana.ai>
Date: Thu, 10 Feb 2022 09:36:44 +0200
Subject: [PATCH] [SW-0] gaudi2: sec: modify u8 value from true/false to 1/0

struct members of hl_rr_eng_select are u8 whereas the current
logic uses true or false. Though this is not a functional issue,
this patch modifies from true/false assignment to 1/0 to avoid
any confusion during code walk-through.

Change-Id: I2cefc838b8f5bf2018e9ad8b6daa56e009c3865a
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 0e49d94..5949a84 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -817,7 +817,7 @@
 	memset(&rr_sel, 0, sizeof(struct hl_rr_eng_select));
 
 	/* Block only pcie RR in ppboot */
-	rr_sel.pcie = true;
+	rr_sel.pcie = 1;
 
 	/* CFG SPACE: 0x1000007FF8000000 - 0x1000007FFFFFFFFF */
 	min = CFG_BASE;
@@ -848,8 +848,8 @@
 	 * the others here, to ensure the config space is blocked before
 	 * reconfiguring in preboot.
 	 */
-	rr_sel.noc = true;
-	rr_sel.psoc = true;
+	rr_sel.noc = 1;
+	rr_sel.psoc = 1;
 
 	/* CFG SPACE: 0x1000007FF8000000 - 0x1000007FFFFFFFFF */
 	min = CFG_BASE;
@@ -859,9 +859,9 @@
 	/*
 	 * As per LBW spec noc and pcie rtrs are reconfigured.
 	 */
-	rr_sel.noc = true;
-	rr_sel.pcie = true;
-	rr_sel.psoc = false;
+	rr_sel.noc = 1;
+	rr_sel.pcie = 1;
+	rr_sel.psoc = 0;
 
 	/*
 	 * PCIE GLOBAL CONF end to CFG SPACE end
@@ -966,7 +966,7 @@
 	memset(&rr_sel, 0, sizeof(struct hl_rr_eng_select));
 
 	/* Only block pcie RR in ppboot */
-	rr_sel.pcie = true;
+	rr_sel.pcie = 1;
 
 	hl_secure_hbw(rr_sel);
 }
@@ -978,8 +978,8 @@
 	memset(&rr_sel, 0, sizeof(struct hl_rr_eng_select));
 
 	/* Block noc and psoc RR in preboot */
-	rr_sel.noc = true;
-	rr_sel.psoc = true;
+	rr_sel.noc = 1;
+	rr_sel.psoc = 1;
 
 	hl_secure_hbw(rr_sel);
 }
@@ -990,7 +990,7 @@
 
 	memset(&rr_sel, 0, sizeof(struct hl_rr_eng_select));
 	/* Block pcie RR for dyn rr */
-	rr_sel.pcie = true;
+	rr_sel.pcie = 1;
 
 	hl_secure_hbw_rr(rr_idx, min64, max64, rr_sel);
 }
@@ -1119,7 +1119,7 @@
 
 	memset(&rr_sel, 0, sizeof(struct hl_rr_eng_select));
 	/* select pcie rr */
-	rr_sel.pcie = true;
+	rr_sel.pcie = 1;
 
 	/*
 	 * Open LBW RR that was closed in ppboot.
@@ -1206,7 +1206,7 @@
 	struct hl_rr_eng_select rr_sel;
 
 	memset(&rr_sel, 0, sizeof(struct hl_rr_eng_select));
-	rr_sel.pcie = true;
+	rr_sel.pcie = 1;
 
 	if (!preboot_security_is_enabled())
 		return;
@@ -1240,7 +1240,7 @@
 	struct hl_rr_eng_select rr_sel;
 
 	memset(&rr_sel, 0, sizeof(struct hl_rr_eng_select));
-	rr_sel.pcie = true;
+	rr_sel.pcie = 1;
 
 	if (!preboot_security_is_enabled())
 		return;
